web/src/beta/features/Visualizer/Crust/Widgets/Widget/builtin/DataAttribution/hooks.ts

```diff
@@ -0,0 +1,95 @@
+import { Credit as ProcessesCredit } from "@reearth/beta/utils/value";
+import { CreditItem, Credits } from "@reearth/core";
+import { useEffect, useMemo, useState } from "react";
+
+import { Widget } from "../../types";
+
+type WidgetCredit = {
+  description?: string;
+  logo?: string;
+  creditUrl: string;
+};
+
+export const useDataAttribution = ({
+  credits,
+  widget
+}: {
+  credits?: Credits;
+  widget: Widget;
+}) => {
+  const [cesiumCredit, setCesiumCredit] = useState<ProcessesCredit | undefined>(
+    undefined
+  );
+  const [otherCredits, setOtherCredits] = useState<
+    ProcessesCredit[] | undefined
+  >(undefined);
+
+  const widgetCredits = useMemo(
+    () => widget?.property.default || [],
+    [widget?.property.default]
+  );
+
+  useEffect(() => {
+    if (credits) {
+      const cesium = parseCreditHtml(credits?.engine?.cesium);
+      setCesiumCredit(cesium);
+
+      const lightboxCredits = credits.lightbox
+        .map(parseCreditHtml)
+        .filter(Boolean) as ProcessesCredit[];
+      const screenCredits = credits.screen
+        .map(parseCreditHtml)
+        .filter(Boolean) as ProcessesCredit[];
+
+      const widgetProcessesCredits: ProcessesCredit[] = widgetCredits
+        .filter(
+          (widgetCredit: WidgetCredit) =>
+            widgetCredit.description ||
+            widgetCredit.logo ||
+            widgetCredit.creditUrl
+        )
+        .map((widgetCredit: WidgetCredit) => ({
```

Comment:
```suggestion
        .map((widgetCredit: WidgetCredit): ProcessesCredit => ({
```
If you define the output type, type checker will help you find the error